### PR TITLE
refactor: remove shelljs and update tests

### PR DESCRIPTION
**Pull request by erisu**
### Motivation and Context

Remove `shelljs` dependency in favor of using `fs-extra`

https://issues.apache.org/jira/projects/CB/issues/CB-14140

### Description

* Removes `shelljs` dependency.
* Added `execa`
* Cleanup test

### Testing

- `npm t`

### Checklist

- [x] I've run the tests to see all new and existing tests pass

### CODE REVIEW

File: tests/spec/unit/lib/update.spec.js [hunk 1]
```diff
@@ -35,12 +35,9 @@ describe('Update', () => {
     describe('help export method', () => {
         it('should warn that updating is not support for Electron.', () => {
             const logSpy = jasmine.createSpy('log');
-            update.__set__('console', {
-                log: logSpy
-            });
+            update.__set__('console', { log: logSpy });
 
             update.help();
-
```
Comment (raphinesse) [inline tests/spec/unit/lib/update.spec.js]: Could we remove all the unrelated changes from this file?

File: tests/spec/unit/templates/cordova/lib/build.spec.js [hunk 2]
```diff
@@ -1736,9 +1735,7 @@ describe('Testing build.js:', () => {
     describe('Module exports help', () => {
         it('should display help usage.', () => {
             const help = build.__get__('module.exports.help');
-
             const argv = { binPath: 'bin' };
-
```
Comment (raphinesse) [inline tests/spec/unit/templates/cordova/lib/build.spec.js]: Could we remove all the unrelated changes from this file?

File: bin/templates/cordova/lib/prepare.js [hunk 3]
```diff
@@ -305,45 +304,35 @@ function createResourceMap (cordovaProject, locations, resources) {
...
 
-/**
- * Get a map containing resources of a specified name (or directory) to the target directory.
- */
-function mapResources (rootDir, sourcePath, targetPath) {
-    const pathMap = {};
-    shell.ls(path.join(rootDir, sourcePath)).forEach(() => {
-        pathMap[sourcePath] = targetPath;
-    });
-    return pathMap;
```
Comment (raphinesse) [inline bin/templates/cordova/lib/prepare.js]: We cannot simply remove this, can we? `mapResources` returned `{ [sourcePath]: targetPath }` for files that existed and `{}` for those that didn't. The new logic behaves as if all files existed. A possible implementation without `shelljs` could be:
```js
return fs.existsFileSync(path.join(rootDir, sourcePath))
    ? { [sourcePath]: targetPath } : {};
```

Comment (erisu) [reply bin/templates/cordova/lib/prepare.js]: I still want to drop the method. 
I added `fs.existsSync` where this method use to be called and added back `{}` though it shouldn't be there. I will be refactoring this area more in a separate PR.

File: tests/spec/unit/templates/cordova/lib/prepare.spec.js [hunk 4]
```diff
@@ -1414,7 +1414,7 @@ describe('Testing prepare.js:', () => {
 
     describe('createResourceMap method', () => {
         let createResourceMap;
-        let shellLsSpy;
+        let existsSyncSpy;
```
Comment (raphinesse) [inline tests/spec/unit/templates/cordova/lib/prepare.spec.js]: Is this still used outside of `beforeEach`?

File: tests/spec/unit/templates/cordova/lib/prepare.spec.js [hunk 5]
```diff
@@ -1610,27 +1590,26 @@ describe('Testing prepare.js:', () => {
...
         beforeEach(() => {
             prepare = rewire(path.resolve(__dirname, '..', '..', '..', '..', '..', '..', 'bin', 'templates', 'cordova', 'lib', 'prepare'));
 
             cordovaProject = Object.assign({}, cordovaProjectDefault);
             mapResources = prepare.__get__('mapResources');
-            shellLsSpy = prepare.__get__('mapResources');
 
-            shellLsSpy = jasmine.createSpy('ls').and.returnValue([true]);
-            prepare.__set__('shell', { ls: shellLsSpy });
+            existsSyncSpy = jasmine.createSpy('existsSyncSpy').and.returnValue([true]);
```
Comment (raphinesse) [inline tests/spec/unit/templates/cordova/lib/prepare.spec.js]:
```suggestion
            existsSyncSpy = jasmine.createSpy('existsSyncSpy').and.returnValue(true);
```

File: tests/spec/unit/templates/cordova/lib/prepare.spec.js [hunk 6]
```diff
@@ -1610,27 +1590,26 @@ describe('Testing prepare.js:', () => {
...
+            prepare.__set__('fs', { existsSync: existsSyncSpy });
         });
 
         it('should not be called if resource does not exist.', () => {
             mapResources(cordovaProject.root, '', '');
 
-            shellLsSpy = jasmine.createSpy('ls').and.returnValue([false]);
-            prepare.__set__('shell', { ls: shellLsSpy });
+            existsSyncSpy = jasmine.createSpy('existsSyncSpy').and.returnValue([false]);
+            prepare.__set__('fs', { existsSync: existsSyncSpy });
```
Comment (raphinesse) [inline tests/spec/unit/templates/cordova/lib/prepare.spec.js]:
```suggestion
```

File: tests/spec/unit/templates/cordova/lib/prepare.spec.js [hunk 7]
```diff
@@ -1610,27 +1590,26 @@ describe('Testing prepare.js:', () => {
...
+            existsSyncSpy = jasmine.createSpy('existsSyncSpy').and.returnValue([true]);
+            prepare.__set__('fs', { existsSync: existsSyncSpy });
         });
 
         it('should not be called if resource does not exist.', () => {
             mapResources(cordovaProject.root, '', '');
 
-            shellLsSpy = jasmine.createSpy('ls').and.returnValue([false]);
-            prepare.__set__('shell', { ls: shellLsSpy });
+            existsSyncSpy = jasmine.createSpy('existsSyncSpy').and.returnValue([false]);
```
Comment (raphinesse) [inline tests/spec/unit/templates/cordova/lib/prepare.spec.js]:
```suggestion
            existsSyncSpy.and.returnValue(false);
```

File: tests/spec/unit/templates/cordova/lib/prepare.spec.js (hunk 4 above)
Comment (erisu) [reply tests/spec/unit/templates/cordova/lib/prepare.spec.js]: removed
